resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -385,6 +385,9 @@ private[spark] class KubernetesClusterSchedulerBackend(
         .withValue(cp)
         .build()
     }
+    val nodeSelector = ConfigurationUtils.parseKeyValuePairs(
```

Comment:
We should use the new key-value pair structure. See how we handle labels and annotations.

`spark.kubernetes.driver.nodeselector.<key>` = <value>`